src/client/js/client.js

```diff
@@ -2015,6 +2017,68 @@ define([
             });
         };
 
+        //reassignGuids - reassigning GUIDs that collide in the given state of the project
+        this.reassignGuids = function (projectId, commitHash, callback) {
+            var command = {};
+            command.command = 'reassignGuids';
+            command.projectId = projectId;
+            command.commitHash = commitHash;
+            logger.debug('reassignGuids, command', command);
+            if (command.projectId && command.commitHash) {
+                storage.simpleRequest(command, function (err, result) {
+                    if (err && !result) {
+                        logger.error('reassignGuids failed with error', err);
+                        callback(err);
+                    } else {
+                        callback(err, result);
+                    }
+                });
+            } else {
+                callback(new Error('invalid parameters!'));
+            }
+        };
+
+        //checking if the import is in the proper format as its intended usage
+        //TODO extra checkings can be done here - now everything is planned to be synchronous
+        this.checkImport = function (jsonImport, importType) {
+            if (!jsonImport) {
+                return 'Import should always be a valid JSON object!';
+            }
+
+            switch (importType) {
+                case CORE_CONSTANTS.EXPORT_TYPE_FULL_PROJECT:
+                    if (jsonImport._metadata && jsonImport._metdata.type) {
+                        if (jsonImport._metadata.type !== CORE_CONSTANTS.EXPORT_TYPE_FULL_PROJECT) {
+                            return 'Import is of type \'library\' and not of \'full project\'!';

```

Comment:
Couldn't this be a method within serialization? The client is pretty packed as it is and this doesn't use the state as far as I can see.